[AUTOCUT] OS Distribution Build Failed for k-NN-3.0.0

***Received Error***: **Error building k-NN, retry with: ./build.sh manifests/3.0.0/opensearch-3.0.0.yml --component k-NN**.
                      The distribution build for k-NN has failed.
                      Please see build log at https://build.ci.opensearch.org/job/distribution-build-opensearch/7056/consoleFull

```
Tests with failures:
 - org.opensearch.knn.index.codec.KNN920Codec.KNN920CodecTests.testMultiFieldsKnnIndex
 - org.opensearch.knn.index.codec.KNN910Codec.KNN910CodecTests.testMultiFieldsKnnIndex
 - org.opensearch.knn.index.codec.KNN950Codec.KNN950CodecTests.testMultiFieldsKnnIndex
 - org.opensearch.knn.index.codec.KNN940Codec.KNN940CodecTests.testMultiFieldsKnnIndex

430 tests completed, 4 failed, 2 skipped

> Task :test FAILED

FAILURE: Build failed with an exception.

```

```
REPRODUCE WITH: ./gradlew ':test' --tests "org.opensearch.knn.index.codec.KNN950Codec.KNN950CodecTests.testMultiFieldsKnnIndex" -Dtests.seed=C5899D3D923ACFD1 -Dtests.security.manager=false -Dtests.locale=el-CY -Dtests.timezone=America/Cuiaba -Druntime.java=17

org.opensearch.knn.index.codec.KNN950Codec.KNN950CodecTests > testMultiFieldsKnnIndex FAILED
    java.lang.AssertionError: expected:<3> but was:<2>
        at __randomizedtesting.SeedInfo.seed([C5899D3D923ACFD1:7FC98B08F9D6F6A2]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.knn.index.codec.KNNCodecTestCase.testMultiFieldsKnnIndex(KNNCodecTestCase.java:155)
        at org.opensearch.knn.index.codec.KNN950Codec.KNN950CodecTests.testMultiFieldsKnnIndex(KNN950CodecTests.java:23)
```

--- a/src/test/java/org/opensearch/knn/index/codec/KNNCodecTestCase.java
+++ b/src/test/java/org/opensearch/knn/index/codec/KNNCodecTestCase.java
@@ -225,7 +225,6 @@ public class KNNCodecTestCase extends KNNTestCase {
         IndexWriterConfig iwc = newIndexWriterConfig();
         iwc.setMergeScheduler(new SerialMergeScheduler());
         iwc.setCodec(codec);
-        iwc.setMergePolicy(NoMergePolicy.INSTANCE);
 
         FieldType fieldType = new FieldType(KNNVectorFieldMapper.Defaults.FIELD_TYPE);
         fieldType.putAttribute(KNNConstants.MODEL_ID, modelId);
@@ -320,7 +319,6 @@ public class KNNCodecTestCase extends KNNTestCase {
         IndexWriterConfig iwc = newIndexWriterConfig();
         iwc.setMergeScheduler(new SerialMergeScheduler());
         iwc.setCodec(codec);
-        iwc.setMergePolicy(NoMergePolicy.INSTANCE);
 
         /**
          * Add doc with field "test_vector_one"

--- a/src/test/java/org/opensearch/knn/index/codec/KNNCodecTestCase.java
+++ b/src/test/java/org/opensearch/knn/index/codec/KNNCodecTestCase.java
@@ -9,6 +9,7 @@ import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat;
 import org.apache.lucene.document.KnnVectorField;
+import org.apache.lucene.index.NoMergePolicy;
 import org.apache.lucene.index.VectorSimilarityFunction;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.TopDocs;
@@ -121,6 +122,7 @@ public class KNNCodecTestCase extends KNNTestCase {
         IndexWriterConfig iwc = newIndexWriterConfig();
         iwc.setMergeScheduler(new SerialMergeScheduler());
         iwc.setCodec(codec);
+        iwc.setMergePolicy(NoMergePolicy.INSTANCE);
 
         /**
          * Add doc with field "test_vector"
@@ -223,6 +225,7 @@ public class KNNCodecTestCase extends KNNTestCase {
         IndexWriterConfig iwc = newIndexWriterConfig();
         iwc.setMergeScheduler(new SerialMergeScheduler());
         iwc.setCodec(codec);
+        iwc.setMergePolicy(NoMergePolicy.INSTANCE);
 
         FieldType fieldType = new FieldType(KNNVectorFieldMapper.Defaults.FIELD_TYPE);
         fieldType.putAttribute(KNNConstants.MODEL_ID, modelId);
@@ -317,6 +320,7 @@ public class KNNCodecTestCase extends KNNTestCase {
         IndexWriterConfig iwc = newIndexWriterConfig();
         iwc.setMergeScheduler(new SerialMergeScheduler());
         iwc.setCodec(codec);
+        iwc.setMergePolicy(NoMergePolicy.INSTANCE);
 
         /**
          * Add doc with field "test_vector_one"

--- a/src/test/java/org/opensearch/knn/index/codec/KNNCodecTestCase.java
+++ b/src/test/java/org/opensearch/knn/index/codec/KNNCodecTestCase.java
@@ -122,6 +122,7 @@ public class KNNCodecTestCase extends KNNTestCase {
         IndexWriterConfig iwc = newIndexWriterConfig();
         iwc.setMergeScheduler(new SerialMergeScheduler());
         iwc.setCodec(codec);
+        // Set merge policy to no merges so that we create a predictable number of segments.
         iwc.setMergePolicy(NoMergePolicy.INSTANCE);
 
         /**

